### DISCUSSION TITLE
`routes/index/index.svelte` => `http://localhost:3000` instead of `http://localhost:3000/index`

### DISCUSSION THREAD

**Discussion by windyclory** [sveltejs/kit#3131] [Ideas]
Hi
Updating this post because previously I didn't explain it well.

---

I have a few components and a few js files that are only related to the home page. In `src/lib` I can create a folder name `home` and place the components and js files in there. The thing that I don't like here is that I don't want to see my home page files every time that I look inside `src/lib`. I want to see those files only when I want to work on the home page.
![Screenshot 2022-01-04 165505](https://user-images.githubusercontent.com/96848024/148144243-94b0722a-4be6-454d-a6c6-ed0ccbf7863e.png)

Instead of creating the `src/lib/home` folder, I can do another thing. I can create a folder in `src/routes/_home` and place the components and js files in there. But it's the same thing, I'm gonna see that `src/routes/_home` folder every time I look at `src/routes`.
![Screenshot 2022-01-04 165731](https://user-images.githubusercontent.com/96848024/148144250-0482591d-fa4b-4df5-9acf-3a05d2afb767.png)

What I want is being able to do this:
![Screenshot 2022-01-04 165948](https://user-images.githubusercontent.com/96848024/148144344-05dfda8a-91fe-44d8-b128-6a9b09a172f2.png)
Without getting this result `localhost:3000/index` instead of `localhost:3000`.

Maybe this isn't right to do, maybe it doesn't even wort the time and effort...🤷‍♀️

**Comment by chespinoza** [comment]
I'm curious about why you'd want to create an `index` folder? index files as in any other web framework and in webdev in general as a convention are rendered as the default file from any folder (exceptions would be things like assets, css, js), this is not only a svelte thing, so it's just following a stablished convention.
Therefore if you need a default file to be rendered in your root web dir ( i.e. http://localhost:3000) you need to have a `index.svelte` under `routes` folder and that's it.

`index/index.svelte` shouldn't map to the root dir (http://localhost:3000) as index folder it's a folder so the current behavior is consistent with how web servers behave in general.

If anyway you'd need to do something like that for some weird reason you'd do it in another layer, for instance using a Reverse Proxy Server like Nginx interfacing Nodejs, in that case it'd be something not related to sveltekit.

**Reply by windyclory** [reply]
I updated the post, please check it out again🌸
